SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -697,25 +699,34 @@ void ActorSystem::ComponentRemoved(const Worker_EntityId EntityId, const Worker_
 
 	if (AActor* Actor = Cast<AActor>(NetDriver->PackageMap->GetObjectFromEntityId(EntityId).Get()))
 	{
-		FUnrealObjectRef ObjectRef(EntityId, ComponentId);
+		const FUnrealObjectRef ObjectRef(EntityId, ComponentId);
 		if (ComponentId == SpatialConstants::DORMANT_COMPONENT_ID)
 		{
 			GetOrRecreateChannelForDormantActor(Actor, EntityId);
 		}
 		else if (UObject* Object = NetDriver->PackageMap->GetObjectFromUnrealObjectRef(ObjectRef).Get())
 		{
-			if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId))
-			{
-				TWeakObjectPtr<UObject> WeakPtr(Object);
-				Channel->OnSubobjectDeleted(ObjectRef, Object, WeakPtr);
+			DestroySubObject(EntityId, Object, ObjectRef);
+		}
+	}
+}
 
-				Actor->OnSubobjectDestroyFromReplication(Object);
+void ActorSystem::DestroySubObject(const Worker_EntityId EntityId, UObject* Object, const FUnrealObjectRef& ObjectRef) const
```

Comment:
I guess this could theoretically be by ref instead of pointer, since we assume it's non-null?
However, it does introduce really weird things further down, since Unreal generally operates on pointers.

Reply:
hmm, well it would mean converting all pointers going in into vals/refs and then back to pointers inside the  function - is the extra safety is worth that?

Reply:
Yeah, that's what I meant by the weird things. I am not the biggest proponent or expert on refs, we could ask Martin,